### ISSUE TITLE
Issue with typescript definitions

### ISSUE THREAD

**Issue by dhovart** [isaachinman/next-i18next#220]
I have set up  next-i18next in my project and it's working fine, but I have an issue with the typescript definitions : my IDE is reporting me the following:

```
 Argument of type 'typeof MyApp' is not assignable to parameter of type 'ComponentType<{}>'.
  Type 'typeof MyApp' is not assignable to type 'ComponentClass<{}, any>'.
    Types of parameters 'props' and 'props' are incompatible.
      Type '{}' is missing the following properties from type 'Readonly<DefaultAppIProps & AppProps<Record<string, string | string[] | undefined>>>': pageProps, Component, router.
```

I've set a custom next App class in `_app.tsx` like in the example here: https://github.com/isaachinman/next-i18next/blob/master/examples/simple/pages/_app.js

The error disappear if I specify `class MyApp extends App<any>` but it doesn't seem to be the correct way of handling the problem to me. I've tried to replace `any` with `AppProps`, `DefaultAppIProps` and an intersection of the two but nothing works.

I'm new to typescript, so I'm not exactly sure what's going on, if it's me doing something wrong or if this is an issue with the way the types are defined.

I'm using version 0.34.0 of next-i18next with Next 8.0.3. Thanks in advance for your help (and for your work on the library).

**Comment by isaachinman** [comment]
@felixmosh Any idea how we can improve here?

**Comment by westpole** [comment]
Recently, I stumbled over the same issue. After some investigation on describe problem, I came across this page https://flow.org/en/docs/react/types/#toc-react-componenttype.
According to the docs:
> React.ComponentType<Props> does not include intrinsic JSX element types like div or span. See React.ElementType if you also want to include JSX intrinsics.

So, I went into the module source code and I made some changes in the `types.d.ts`. 

```js
appWithTranslation(Component: React.ComponentType | React.ElementType): any;
```
and now Typescript does not complain about the problem.

Unfortunately, I did not test this solution as I just started to implement translation in my React app.

@dhovart @isaachinman could you guys check if provided solution is correct? Thank you.

**Comment by dhovart** [comment]
@westpole That seems to fix the problem with me. You should make a PR with this change.

**Comment by isaachinman** [comment]
Those are flow docs, not TypeScript. Regardless, `appWithTranslation` is a HOC so while it definitely does take a React component, it also returns one. I'd prefer to declare types if possible, rather than `any`.

I don't actually use this package with any TypeScript projects (yet), and will accept any sensible PRs.

**Comment by westpole** [comment]
@isaachinman you are right, but Flow has a good documentation for any type that is included in a @types/react, so I thought it could helpful to get to know the issue.

Also, may be you could help me, I can't commit changes to my local repo. Command `yarn run build:example` is failing with an error `ModuleNotFoundError: Module not found: Error: Can't resolve 'next-i18next/dist/commonjs'` (it fails during server generation: building (13%) 29/40 modules 11 active
  ../../node_modules/@babel/runtime-corejs2/core-js/object/create.js).  If that is a known issue, could you provide any description? If it is new, then I will create a new issue. Thanks.

**Comment by isaachinman** [comment]
@westpole Yup, it's definitely still helpful. This is a simple typing issue that should be extremely simple to fix.

> ModuleNotFoundError: Module not found: Error: Can't resolve 'next-i18next/dist/commonjs'

Yes, I suppose this is "new", but it's just a DX thing. Can you try doing a `yarn run-example` first and see if that fixes the issue? Please do open a new issue, and submit a PR if you'd like. We need an npm script like `postinstall` that does a proper setup for the example dirs, but only when in dev, and not when downloading the package from registries.

**Comment by westpole** [comment]
Oh, I see `yarn run-example` runs `yarn install` that fixed the problem that I had before.
I guess build process is missing `yarn install` that should install dependencies in the example folder.

**Comment by isaachinman** [comment]
@westpole Yeah that's correct, all though it would need to be a bit more nuanced than that.

**Comment by felixmosh** [comment]
@isaachinman, checking that.

**Comment by isaachinman** [comment]
Fixed by #225.
